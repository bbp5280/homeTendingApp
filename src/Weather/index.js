import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { fetchWeather } from '../api/api';
import { WeatherCard } from '../WeatherCard/index';


class Weather extends Component {
  constructor(){
    super();
    this.state = {
      weatherForcast: []
    };
  }

  async componentDidMount(){
    await this.fetchSevenDayForcast();
  }

  async fetchSevenDayForcast () {
    let address = this.props.message.address;
    let zip = address.substr(address.length - 5);
    let sevenDayForcast = await fetchWeather(zip);
    this.setState({
      weatherForcast: sevenDayForcast
    });
  }

  buildWeatherCards (forcast) {
    return forcast.map((day, index) => {
      return <WeatherCard high={day.high.fahrenheit}
        low={day.low.fahrenheit}
        conditions={day.conditions}
        icon={day.icon_url}
        day={day.date.weekday}
        key={`${day.low.fahrenheit}${day.high.fahrenheit}${index}`}
      />;
    });
  }

  render(){
    return (
      <div className='outer-weather-modal'>
        <div className='inner-weather-modal'>
          <h2>7 Day Forcast</h2>
          <div className='weather-card-wrapper'>
            {this.buildWeatherCards(this.state.weatherForcast)}
          </div>
          <button onClick={this.props.close} className='close-button'>Close</button>
        </div>
      </div>
    );
  }
}

Weather.propTypes = {
  message: PropTypes.object,
  close: PropTypes.func
};

export default Weather;
