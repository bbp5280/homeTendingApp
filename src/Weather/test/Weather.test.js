import { shallow, configure } from 'enzyme';
import  Weather from '../index';
import React from 'react';
import  Adapter from 'enzyme-adapter-react-15';

configure({ adapter: new Adapter() });

describe('CardCatalog snapshot', () => {

  it('should always match the snapshot', () => {
    const mkFun = jest.fn();
    const wrapper = shallow(<Weather
      message={[{houseid: 1,
        message:'Message'}]}
      close={mkFun}/>);

    expect(wrapper).toMatchSnapshot();

  });
});
