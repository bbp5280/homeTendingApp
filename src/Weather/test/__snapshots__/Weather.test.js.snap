exports[`CardCatalog snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Weather
    close={[Function]}
    message={
        Array [
            Object {
              "houseid": 1,
              "message": "Message",
            },
          ]
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="inner-weather-modal">
        
</div>,
      "className": "outer-weather-modal",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "inner-weather-modal",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="inner-weather-modal">
          
</div>,
        "className": "outer-weather-modal",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "inner-weather-modal",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
